In [5]:
import pandas as pd
import numpy as np
import requests

# contest.list

Returns information about all available contests.

### Parameter	Description
gym	Boolean. If true — than gym contests are returned. Otherwide, regular contests are returned.
Return value: Returns a list of Contest objects. If this method is called not anonymously, then all available contests for a calling user will be returned too, including mashups and private gyms.

In [35]:
data=requests.get("https://codeforces.com/api/contest.list?").json()
data1=pd.DataFrame(data['result'])
data1

id                                   name  type     phase  frozen  \
0     1827              Codeforces Round (Div. 1)    CF    BEFORE   False   
1     1828              Codeforces Round (Div. 2)    CF    BEFORE   False   
2     1824              Codeforces Round (Div. 1)    CF    BEFORE   False   
3     1825              Codeforces Round (Div. 2)    CF    BEFORE   False   
4     1829          Codeforces Round 871 (Div. 4)  ICPC    BEFORE   False   
...    ...                                    ...   ...       ...     ...   
1733     5                Codeforces Beta Round 5  ICPC  FINISHED   False   
1734     4  Codeforces Beta Round 4 (Div. 2 Only)  ICPC  FINISHED   False   
1735     3                Codeforces Beta Round 3  ICPC  FINISHED   False   
1736     2                Codeforces Beta Round 2  ICPC  FINISHED   False   
1737     1                Codeforces Beta Round 1  ICPC  FINISHED   False   

      durationSeconds  startTimeSeconds  relativeTimeSeconds  
0                7200        1684074900             -1039650  
1                7200        1684074900             -1039650  
2                7200        1683547500              -512250  
3                7200        1683547500              -512250  
4                8100        1683383700              -348450  
...               ...               ...                  ...  
1733             7200        1269100800            413934451  
1734             7200        1268395200            414640051  
1735             7200        1267963200            415072051  
1736             7200        1267117200            415918051  
1737             7200        1266580800            416454451  

[1738 rows x 8 columns]

# contest.ratingChanges
Returns rating changes after the contest.

## Parameter	Description
contestId (Required)	Id of the contest. It is not the round number. It can be seen in contest URL.   
For example: /contest/566/status
Return value: Returns a list of RatingChange objects.  

Example: https://codeforces.com/api/contest.ratingChanges?contestId=566

Note: 760*1737 *1141= 1.5 GB

In [41]:
data=requests.get("https://codeforces.com/api/contest.ratingChanges?contestId=5").json()
data1=pd.DataFrame(data['result'])
data1

contestId              contestName       handle  rank  \
0            5  Codeforces Beta Round 5         Petr     1   
1            5  Codeforces Beta Round 5      ilyaraz     2   
2            5  Codeforces Beta Round 5      RAVEman     3   
3            5  Codeforces Beta Round 5       vlad89     4   
4            5  Codeforces Beta Round 5      ilyakor     5   
..         ...                      ...          ...   ...   
327          5  Codeforces Beta Round 5  khaled.hady   315   
328          5  Codeforces Beta Round 5        AnhDQ   315   
329          5  Codeforces Beta Round 5          GDR   315   
330          5  Codeforces Beta Round 5    Batyr_004   315   
331          5  Codeforces Beta Round 5          ika   315   

     ratingUpdateTimeSeconds  oldRating  newRating  
0                 1269108000       1775       1891  
1                 1269108000       1500       1720  
2                 1269108000       1655       1813  
3                 1269108000       1644       1803  
4                 1269108000       1762       1876  
..                       ...        ...        ...  
327               1269108000       1500       1391  
328               1269108000       1500       1391  
329               1269108000       1357       1276  
330               1269108000       1384       1298  
331               1269108000       1610       1476  

[332 rows x 7 columns]

{'status': 'FAILED',
 'comment': "contestId: Rating changes are unavailable, because the contest isn't finished yet"}